views/js/e2e/runner/navigation/linearTestNavigation.spec.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+import '../_routes/runnerRoutes';
+import '../_routes/testExecutionRoutes';
+
+import '../_setup/setupCommands';
+import '../_cleanup/cleanupCommands';
+
+import runnerUrls from '../_urls/runnerUrls';
+import setupSelectors from '../_setup/setupSelectors';
+
+describe('Deliveries', () => {
+
+    /**
+     * Setup to have a proper delivery:
+     * - Start server
+     * - Add necessary routes
+     * - Admin login
+     * - Import and publish e2e example test
+     * - Set guest access on delivery and save
+     * - Logout
+     * - Guest login
+     * - Start Test
+     */
+    beforeEach(() => {
```

Comment:
You could consider using `before` and `after` instead of `beforeEach` and `afterEach`. It's a trade-off. You gain more isolation by running the whole setup and cleanup multiple times, but it would be faster if you had a single test containing all the assertions for the first, second and last items.

Reply:
What I meant was, use `before` and `after` just for the publishing and deletion, but use `beforeEach` to start each new delivery, etc.

Reply:
yes, agree.

Reply:
yes, good point